Chapter1/3_URLify/URLify.py

```diff
@@ -2,21 +2,8 @@
 import unittest
 
 
-def urlify(string, length):
-    '''function replaces single spaces with %20 and removes trailing spaces'''
-    new_index = len(string)
-
-    for i in reversed(range(length)):
-        if string[i] == ' ':
-            # Replace spaces
-            string[new_index - 3:new_index] = '%20'
-            new_index -= 3
-        else:
-            # Move characters
-            string[new_index - 1] = string[i]
-            new_index -= 1
-
-    return string
+def urlify(string, str_len):
+    return "%20".join(string[:str_len].split(' '))
```

Comment:
While I think this could work, we have recently included another one-liner "pythonic" solution:
`text.rstrip().replace(" ", "%20")`